src/esm.ts

```diff
@@ -332,3 +340,19 @@ export function createEsmHooks(tsNodeService: Service) {
     return { source: emittedJs };
   }
 }
+
+class MappedModuleNotFound extends Error {
+  // Same code as other module not found errors.
+  static code = 'ERR_MODULE_NOT_FOUND';
+
+  constructor(specifier: string, base: string, candidates: string[]) {
+    super(
+      [
+        `Cannot find '${specifier}' imported from ${base} using TypeScript path mapping`,
+        'Candidates attempted:',
+        ...candidates.map((candidate) => `- ${candidate}`),
+      ].join('\n')
+    );
+    this.name = `Error [${MappedModuleNotFound.code}]`;
+  }
+}
```

Comment:
Is there any benefit to setting a ts-node specific `code` on these errors?  `"ERR_TS_NODE_MAPPED_MODULE_NOT_FOUND"`?  Or maybe make this a subclass of `AggregateError`?

We don't need to do this now.  I'll keep it in mind for the future.

Reply:
I realized that node itself throws multiple flavors of `ERR_MODULE_NOT_FOUND` with different phrasing for the error message.  So I think it would be reasonable for us to use the same error code for the errors we throw.  We can also set `.errors` to an array of the individual errors to match the shape of `AggregateError` instances.

I'm happy to implement this change.

Reply:
...and I just realized you already have the `code` there, nice.  I tweaked it to set `code` on the error instances, too.